### PR TITLE
Feature/boilerplate generation

### PR DESCRIPTION
**Pull request by cadmiumcat**
- Adds boilerplate for API
- updates dependencies, go version and adds vulnerability to ignore
- Uses [log.go/v2 ](https://github.com/ONSdigital/log.go/releases/tag/v2.0.0)

### CODE REVIEW

File: dp-collection-api.nomad [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    restart {
+      attempts = 3
+      delay    = "15s"
+      interval = "1m"
+      mode     = "delay"
+    }
+
+    task "dp-collection-api-web" {
```
Comment (nshumoogum) [inline dp-collection-api.nomad:29]: Is there a need for the collection API to be run in web as well as publishing?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: dp-collection-api.nomad (hunk 1 above)
Comment (cadmiumcat) [reply dp-collection-api.nomad:29]: But I think we did decide to start just with `publishing` as we can change it later. So I'll go ahead and remove this

File: ci/build.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+
+platform: linux
+
+image_resource:
+  type: docker-image
+  source:
+    repository: golang
+    tag: 1.16.3
```
Comment (sandypadmanabhan) [inline ci/build.yml:9]: A new patch version 1.16.4 has been released which includes security fix and to net/http packages. Will it work to our advantage?

File: dp-collection-api.nomad [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        }
+      }
+
+      template {
+        source      = "${NOMAD_TASK_DIR}/vars-template"
+        destination = "${NOMAD_TASK_DIR}/vars"
+      }
+
+      vault {
+        policies = ["dp-collection-api-publishing"]
```
Comment (sandypadmanabhan) [inline dp-collection-api.nomad:72]: as this app is run just in publishing the policies name should just be `dp-collection-api`. Only if the app runs in both we append web/publishing to the name.

Comment (cadmiumcat) [reply dp-collection-api.nomad:72]: we've decide to keep this because it is unclear whether we'll need the app to run in `web` as well

Comment (sandypadmanabhan) [reply dp-collection-api.nomad:72]: Currently we have just the publishing group, and the web group has already been deleted, I would personally stick same style we use for other app's. If we add web, then we can always update it. But, I don't mind if the decision is made consciously. I am happy for the PR to move forward

Comment (cadmiumcat) [reply dp-collection-api.nomad:72]: Yeah, the decision was definitely made conciously 🙂

**Review by DavidSubiros — APPROVED** [review]
- Code makes sense
- Unit test and component tests pass
- Run `make debug` and validated `http://localhost:26000/hello` and `http://localhost:26000/health` work as expected
- Checked that previous review comments have been addressed or property discussed.

Just a note for the future: we might consider changing the service package to simplify the code, similar to Filter API, for example. This is just a suggestion for the future, not a change request for this PR.

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**